bridge/src/main/java/io/stargate/bridge/service/interceptors/NewConnectionInterceptor.java

```diff
@@ -77,37 +81,38 @@ public <ReqT, RespT> Listener<ReqT> interceptCall(
     try {
       Context context = Context.current();
 
-      String token = headers.get(TOKEN_KEY);
-      if (token == null) {
-        call.close(Status.UNAUTHENTICATED.withDescription("No token provided"), new Metadata());
-        return new NopListener<>();
+      if (shouldCreateConnection(call)) {
```

Comment:
Looks like the logic for "shouldCreateConnection()" is bit heavy-weight, and in 99% of cases would return `true`, to require non-null token.
Given this, would this make sense to swap the logic to only check if connection-create is needed if `token` is `null`? To distinguish between error (rare) and GetSupportedFeatures (rare as well).

Reply:
Branch prediction ;)

Reply:
And that `equals` should not be heavy at all.. Maybe I am missing something..

Reply:
I guess we could rewrite the code like this:
```java
if (token == null) {
  if (shouldCreateConnection(call)) {
    // error out and return
  }
  // fallthrough to return statement
} else {
  // create and attach connection
}
return Contexts.interceptCall(context, call, headers, next);
```
Then we are not completely skipping auth for GetSupportedFeatures, just making it optional: if a token is present, we still create a connection. It won't be used for this call but that does not really matter, since connections are cached.

But I'm not sure it's worth it: as Ivan said the check is not expensive, and I find the original version easier to follow.